core/src/main/java/org/opensearch/sql/expression/datetime/DateTimeFunction.java

```diff
@@ -761,7 +768,7 @@ private ExprValue exprDayName(ExprValue date) {
   /**
    * Day of Month implementation for ExprValue.
    *
-   * @param date ExprValue of Date/String type.
+   * @param date ExprValue of Date/Datetime/String/Time/Timestamp type.
```

Comment:
Does `date.dateValue()` work with argument of type `STRING`?

Reply:
Yes. `day_of_month('2023-01-03')` and `day_of_month("2023-01-03")` both work.